src/vector/vector_tests/mod.rs

```diff
@@ -491,4 +491,30 @@ mod tests {
         assert_eq!(ft.field("Value").unwrap().unwrap().into_real(), Some(45.78));
         assert_eq!(ft.field("Int_value").unwrap().unwrap().into_int(), Some(1));
     }
+
+    #[test]
+    fn test_features_reset() {
+        with_layer("roads.geojson", |mut layer| {
+            assert_eq!(layer.features().count(), layer.features().count(),);
+        });
+    }
+
+    #[test]
+    fn test_features_aliasing_compile_fail() {
+        let t = trybuild::TestCases::new(); // A compilation test that should fail.
```

Comment:
Cool to see `trybuild`! 

I've never seen tests like this outside of the actual compiler dev, but it makes sense!

Reply:
It's a bit of a heavy dep., but seems useful for this crate as we try to impose rust semantics on ffi libs.  I figure we could always comment it out, and remove the deps if CI takes too long.